### PR TITLE
Add github security advisory

### PR DESCRIPTION
**Pull request by masahiro331**
Add github security advisories. 

related PR: https://github.com/aquasecurity/vuln-list-update/pull/29
related Issue: https://github.com/aquasecurity/trivy/issues/344, https://github.com/aquasecurity/trivy/issues/415

TODO: 
- [x] Add GHSA db 
- [x] Add tests
- [x] Add vuln-list-update
- [x] Test Trivy scan

### CODE REVIEW

File: pkg/vulnsrc/ghsa/ghsa.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		for _, identifier := range ghsa.Advisory.Identifiers {
+			if identifier.Type == "CVE" {
+				vulnId = identifier.Value
+			}
+		}
+
+		for index, patchVersion := range pvs {
+			// e.g. GHSA-r4x3-g983-9g48 PatchVersion has "<" operator
+			if strings.HasPrefix(patchVersion, "<") {
+				avs[index] = fmt.Sprintf("%s, %s", avs[index], pvs[index])
```
Comment (simar7) [inline pkg/vulnsrc/ghsa/ghsa.go]: Is it possible that `avs` is shorter in length than `pvs` and doesn't have `index` number of elements for this to panic?

Comment (masahiro331) [reply pkg/vulnsrc/ghsa/ghsa.go]: Fixed code!

**Comment by mrueg** [comment]
@masahiro331 @knqyf263 can you get the update of go.mod https://github.com/aquasecurity/trivy-db/pull/28/commits/7be4333b86c674593f276742da6cce70c418fb00 in independently of that PR?

**Comment by simar7** [comment]
hey @masahiro331 and @knqyf263 so if I understand correctly, all we need to do is to build a new db with this branch, use it with Trivy before we can merge this PR?

**Comment by masahiro331** [comment]
@simar7 Please wait a few days.
I am now counting the duplicate of vulnerabilities in the db.

**Comment by knqyf263** [comment]
Hi @simar7 , would you review it?

File: pkg/vulnsrc/ghsa/ghsa.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		vulnId := ghsa.Advisory.GhsaId
+		for _, identifier := range ghsa.Advisory.Identifiers {
+			if identifier.Type == "CVE" {
+				vulnId = identifier.Value
+				foo = append(foo, identifier.Value)
+			}
+		}
+		if len(foo) > 1 {
+			log.Fatal(ghsa.Advisory.GhsaId)
+		}
```
Comment (simar7) [inline pkg/vulnsrc/ghsa/ghsa.go:132]: Was `foo` used for debugging?

File: pkg/vulnsrc/ghsa/ghsa.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	case "LOW":
+		return types.SeverityLow
+	case "MODERATE":
+		return types.SeverityMedium
+	case "HIGH":
+		return types.SeverityHigh
+	case "CRITICAL":
+		return types.SeverityCritical
+	default:
+		log.Fatal("unknown severity")
```
Comment (simar7) [inline pkg/vulnsrc/ghsa/ghsa.go:205]: Should we fatal out or just return the unknown severity?

File: pkg/vulnsrc/ghsa/ghsa.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	case Npm:
+		return "Npm"
+	case Nuget:
+		return "Nuget"
+	case Pip:
+		return "Pip"
+	case Rubygems:
+		return "Rubygems"
+	}
+	log.Fatal("Unknown Ecosystem")
```
Comment (simar7) [inline pkg/vulnsrc/ghsa/ghsa.go:47]: ditto

File: go.mod [hunk 5]
```diff
@@ -4,16 +4,19 @@ go 1.13
 
 require (
 	github.com/BurntSushi/toml v0.3.1
-	github.com/aquasecurity/trivy v0.1.6
```
Comment (simar7) [inline go.mod:7]: Just curious what was this for?

Comment (masahiro331) [reply go.mod:7]: This change.
https://github.com/aquasecurity/trivy-db/commit/c9031b042c04d039492896a9b8d0bca7a97e5d30#diff-3661dc32b6bde1b0939ad409364e48fcL11

Comment (simar7) [reply go.mod:7]: Thanks! I was just curious.

**Comment by knqyf263** [comment]
@simar7 Thank you for the review.
@masahiro331 I'll have a look at last. Just a moment.

**Pull request merged**